src/runtime/refiner.ts

```diff
@@ -368,7 +384,7 @@ export class BinaryExpression extends RefinementExpression {
     if (sp) {
       return sp;
     }
-    if (this.rightExpr instanceof FieldNamePrimitive) {
+    if (this.leftExpr.toString() < this.rightExpr.toString() ) {
```

Comment:
This seems odd ... why is string lexical comparison used to reorder the expression?

Reply:
We just need a total order on expressions (I was considering using a different order, such as the number of field name primitives in the expression, but then we also have to compare those primitives and even then this doesn't correctly rearrange cases like `-2x == x` and `x == -2x`. The string of the expression holds all the expression's information and is total, so it seemed like a good candidate, but I'm open to using something that has nicer properties.

Reply:
Sounds fair - but definitely worth a comment :)